### DISCUSSION TITLE
"First class" CRDT support in Drift

### DISCUSSION THREAD

**Discussion by dkbast** [simolus3/drift#3053] [Ideas]
Since Drift can talk to both sqlite and postgres, it would be great to make the two talk with each other - I understand that @simolus3 has stated before that drift should work well with sync libraries but not provide it out of the box. 

Since there already is a drift_crdt implementation (eg discussed here https://github.com/simolus3/drift/discussions/2725) it would be great if the "remaining obstacles" could be addressed. 

I personally don't feel comfortable introducing drift_crdt as long as I need to manually edit generated code. I think @JanezStupar already did some great work, but maybe it would be something that could also live under the drift umbrella, making sure that it works with both postgres and sqlite backends? The sync mechanism is outside of the scope as it is maintained as part of https://pub.dev/packages/crdt_sync

**Comment by JanezStupar** [comment]
I would love if @simolus3 would be open to this. I am even willing to donate the PRs for the cause.

It would be great however if Simon would provide some guidance and boundaries so that the implementation does not step on any toes.

**Reply by simolus3** [reply]
I think CRDT support is not really in scope for this project, but I'm open to lightweight changes that make it easier to support CRDT structures in a third-party plugin.

Is the generated migration code with the table structures not matching the expected schema in the database the only problem? Changing `drift_dev` to be aware of `drift_crdt` is not really something I'm comfortable with, but maybe there's something that can be done at runtime to e.g. wrap the `TableInfo` classes for the migrator to automatically add the relevant CRDT state columns?

**Reply by JanezStupar** [reply]
@simolus3 I will check your suggestion out and report back. I will try to do that in the next 2 weeks or so.
